core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerClient.scala

```diff
@@ -63,8 +73,26 @@ class DockerClient(dockerHost: Option[String] = None)(executionContext: Executio
     Seq(dockerBin) ++ host
   }
 
-  def run(image: String, args: Seq[String] = Seq.empty[String])(implicit transid: TransactionId): Future[ContainerId] =
-    runCmd((Seq("run", "-d") ++ args ++ Seq(image)): _*).map(ContainerId.apply)
+  def run(image: String, args: Seq[String] = Seq.empty[String])(
+    implicit transid: TransactionId): Future[ContainerId] = {
+    runCmd((Seq("run", "-d") ++ args ++ Seq(image)): _*)
+      .map {
+        ContainerId(_)
+      }
+      .recoverWith {
+        // https://docs.docker.com/v1.12/engine/reference/run/#/exit-status
+        // Exit code 125 means an error reported by the Docker daemon.
+        // Examples:
+        // - Unrecognized option specified
+        // - Not enough disk space
+        case pre @ (_: ProcessRunningException) if pre.exitCode == 125 =>
```

Comment:
`case pre: ProcessRunningException if pre.exitCode == 125`?

Reply:
You are so right.